File: csharp/Synurang/PluginError.cs
using System;

namespace Synurang;

public class PluginError : Exception
{
    public PluginError(string message) : base(message) { }
    public PluginError(string message, Exception inner) : base(message, inner) { }
}

public class PluginClosedError : PluginError
{
    public PluginClosedError() : base("plugin is closed") { }
}

File: java/core/src/main/java/io/github/ivere27/synurang/PluginError.java
package io.github.ivere27.synurang;

/**
 * Exception thrown when a plugin operation fails.
 */
public class PluginError extends Exception {
    public PluginError(String message) {
        super(message);
    }

    public PluginError(String message, Throwable cause) {
        super(message, cause);
    }

    /**
     * Thrown when operations are attempted on a closed plugin.
     */
    public static class ClosedError extends PluginError {
        public ClosedError() {
            super("plugin is closed");
        }
    }
}
